### ISSUE TITLE
Doesn't work for some epub files

### ISSUE THREAD

**Issue by TaylanTatli** [johnfactotum/foliate#536]
**Describe the bug**
When i try to open some epub files or turn pages, foliate can't show file. UI works but doesn't show content. Changin pages does nothing.

**To Reproduce**
1. Open an epub file.
2. Press `N` to go to next page
3. See error

**Expected behavior**
Should see the file content and go to next page.

**Screenshots**
![2020 07 22-16 53 51](https://user-images.githubusercontent.com/754514/88184893-f4c66180-cc3b-11ea-9e60-3cddf31cdb4b.png)

**Version:**
 - Foliate version: 2.4.2
 - OS/Distribution and version: Arch
 - Desktop environment: KDE Plasma 5.19, also same for Openbox, i3 etc.
 - Installation method: Pacman package manager, also tried git version with AUR.

**Additional context**
Here is terminal output:
```
com.github.johnfactotum.Foliate

(com.github.johnfactotum.Foliate:36498): Gtk-WARNING **: 16:47:48.015: Theme parsing error: gtk-dark.css:8378:11: 'max-width' is not a valid property name

(com.github.johnfactotum.Foliate:36498): Gtk-WARNING **: 16:47:48.015: Theme parsing error: gtk-dark.css:8380:7: 'width' is not a valid property name

(com.github.johnfactotum.Foliate:36498): Gtk-WARNING **: 16:47:48.164: Theme parsing error: gtk.css:8378:11: 'max-width' is not a valid property name

(com.github.johnfactotum.Foliate:36498): Gtk-WARNING **: 16:47:48.164: Theme parsing error: gtk.css:8380:7: 'width' is not a valid property name

(WebKitWebProcess:2): Gtk-WARNING **: 16:47:48.605: Theme parsing error: gtk-dark.css:8378:11: 'max-width' is not a valid property name

(WebKitWebProcess:2): Gtk-WARNING **: 16:47:48.605: Theme parsing error: gtk-dark.css:8380:7: 'width' is not a valid property name
free(): invalid pointer
```

**Comment by stevenpusser** [comment]
Are there any public domain or otherwise free epubs that have the issue that we could try testing ourselves?

**Comment by TaylanTatli** [comment]
> Are there any public domain or otherwise free epubs that have the issue that we could try testing ourselves?

Sorry i forgot it. Here an [online epub](https://saltonline.org/media/files/sarkis_ve_when_attitudes_become_form-tutumlar_bicime_donusunce.epub). or [this one](https://www.dropbox.com/s/plbqptvsevehgy3/KPSS%20Tarih_Epub3.epub?dl=0) that i made.

**Comment by fire-eggs** [comment]
The link to your created epub doesn't work. 

However, I checked "Sarkis ..." with epubcheck 4.2.4, and it reported 20 errors and 11 warnings. So I'd not be surprised if Foliate / epub.js has problems with it. 

Here are the results:
[errs.txt](https://github.com/johnfactotum/foliate/files/5176097/errs.txt)

"Sarkis..." identifies itself as Epub 2.0.1 conformant, it's possible it is following a newer standard, but that would also be a flaw with the epub file.

**Comment by johnfactotum** [comment]
The second link indeed doesn't work. But I can open the first file fine. Screenshot:
![image](https://user-images.githubusercontent.com/50942278/92304109-af9b8c00-ef6a-11ea-9d80-20852dd96221.png)

**Comment by fire-eggs** [comment]
Yup, and paging through it, I see no obvious issues.

**Comment by tummychow** [comment]
i can reproduce a segfault consistently with pretty much any fanficfare epub, eg https://gist.github.com/tummychow/366eff2cab22ca9db18f9d702662eb8d/raw/106df6fa52ea1b4eb9e60ef98e6d0979a69c8269/Anonymous%2520-%2520test%2520test%2520(ao3_26379688).epub (https://archiveofourown.org/works/26379688 - link is sfw). i'm also on arch (foliate 2.4.2, gjs 1.66.1, mozjs 78.3.1). any way i can help with the investigation here?

**Comment by fire-eggs** [comment]
>i can reproduce a segfault consistently with pretty much any fanficfare epub

This is interesting. [epubcheck](https://github.com/w3c/epubcheck) 4.2.4 complains about both files, but I had no problems viewing both files with foliate / Linux Mint.

[epubcheck said "test test" had 25 errors and 4 warnings; "ao3_26379688" had 47 errors and 1 warning. So technically both files don't conform to the EPUB specs. I see that both files were generated using Calibre, which I've had problems with in the past].

**Comment by tummychow** [comment]
okay, that helps. i have a feeling the issue was with the lib versions that arch was building with. what versions of gjs/mozjs was your foliate built against on mint?

**Comment by tummychow** [comment]
yep, it's some kind of version conflict. if i install foliate from the arch archive (https://archive.archlinux.org/repos/2020/07/14/), it runs without issue. the 2020-07-14 archive is on foliate 2.4.2, but with gjs 1.64.4 and mozjs 68.10.0.

i can open TaylanTatli's earlier link (sarkis_ve...epub) even with the current arch versions. however, the epubs i linked can only be opened if i downgrade. anyone want to try reproing on another distro? looks like ubuntu groovy is on mozjs 78.

**Comment by fire-eggs** [comment]
I installed foliate as a flatpak.

This might be my ignorance showing, but I don't know why gjs or mozjs are mentioned. I don't recall seeing them as dependencies for foliate. E.g. I don't seem to have gjs installed on my Linux Mint box.

I have this version of webkit:
Gir1.2-webkit2-4.0
2.28.4-0ubuntu0.20.04.1

Something specific I can look for? [and how?]

**Comment by tummychow** [comment]
oh okay, then your gjs is inside the flatpak runtime, which means you're also on 1.64.4 (https://github.com/johnfactotum/foliate/blob/2.4.2/com.github.johnfactotum.Foliate.json#L4 -> https://gitlab.gnome.org/GNOME/gnome-build-meta/-/tags/3.36.7 -> https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/e32210f78e048e46f10b10a7ee0b4add3190ad3c/elements/sdk/gjs.bst#L4). i don't have flatpak set up, but i bet the bug would reproduce if the flatpak was rebuilt for the 3.38 runtime, which uses gjs 1.66.0

**Comment by johnfactotum** [comment]
@tummychow See #608

**Comment by tummychow** [comment]
foliate 2.5.0 does indeed fix the segfault with gjs 1.66 for me. thanks for the quick turnaround
